library/objective-c/Envoy.mm

```diff
@@ -0,0 +1,45 @@
+#import "library/objective-c/Envoy.h"
+
+#import "library/common/main_interface.h"
+
+
+@interface Envoy ()
+@property (nonatomic, strong) NSThread *runner;
+@end
+
+@implementation Envoy
+
+@synthesize runner;
+
+// Create a new Envoy instance. The Envoy runner NSThread is started as part of instance
+// initialization with the configuration provided.
+- (id)initWithConfig:(NSString *)config {
```

Comment:
```suggestion
- (instancetype)initWithConfig:(NSString *)config {
```

Reply:
oo that's right, things have gotten fancier :)

Reply:
I should be calling super init as well, huh

Reply:
Heh